### PR TITLE
Modify mailer configuration requirements

### PR DESCRIPTION
**Pull request by whit537**
This is based on my attempts to get the mailer working for the first time on #598.

### CODE REVIEW

**Comment by whit537** [comment]
Marking this wip, breakage on ldap again ...

**Comment by whit537** [comment]
Okay, let's try that.

File: tools/c7n_mailer/c7n_mailer/handle.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def bootstrap():
     os.environ['PYTHONPATH'] = "%s:%s" % (task_dir, os.environ.get('PYTHONPATH', ''))
     with open(os.path.join(task_dir, 'config.json')) as fh:
         config = json.load(fh)
-    if 'ldap_bind_password' in config:
+    if config['ldap_bind_password']:
```
Comment (kapilt) [inline tools/c7n_mailer/c7n_mailer/handle.py]: if your removing this key from required, then this could be a keyerror.

Comment (kapilt) [inline tools/c7n_mailer/c7n_mailer/handle.py]: nevermind, i see its in defaults now.

**Comment by whit537** [comment]
Rebased, was 3eed662814d1590ff385a2f0df7fb68233ff5089. Looking into `from_address` ...

**Comment by whit537** [comment]
> although we should still retail a default for from_address, as its [optional on the notify action](https://github.com/whit537/cloud-custodian/blob/5d9c785eac28f94d135cba371f8db247c68f9adb/tools/c7n_mailer/c7n_mailer/processor.py#L174).

Kind of. If I'm not mistaken, any from address needs to be verified in one's own SES before sending with it, so the default isn't actually usable. Seems more honest to make it a required field in mailer config, no?

**Comment by kapilt** [comment]
hmm yeah it does re required in mailer config, except when we use notify for data use cases it becomes a dead chicken, but that's really speaking to the want of a separate data relay. anyways, filing that as separate issue

**Comment by whit537** [comment]
Rebased, was 5d9c785eac28f94d135cba371f8db247c68f9adb.

**Pull request merged**